src/MonitorProfiler/Communication/IpcCommServer.cpp

```diff
@@ -0,0 +1,130 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "IpcCommClient.h"
+#include "IpcCommServer.h"
+
+IpcCommServer::IpcCommServer() : _shutdown(false)
+{
+}
+
+IpcCommServer::~IpcCommServer()
+{
+    _domainSocket.~SocketWrapper();
+    std::remove(_rootAddress.c_str());
+}
+
+HRESULT IpcCommServer::Bind(const std::string& rootAddress)
+{
+    if (_shutdown.load())
+    {
+        return E_UNEXPECTED;
+    }
+
+    if (_domainSocket.Valid())
+    {
+        return E_UNEXPECTED;
+    }
+
+    _rootAddress = rootAddress;
+
+    sockaddr_un address;
+    ZeroMemory(&address, sizeof(address));
+
+    if ((rootAddress.length() == 0) || (rootAddress.length() >= sizeof(address.sun_path)))
+    {
+        return E_INVALIDARG;
+    }
+
+    //We don't error check this on purpose
+    std::remove(rootAddress.c_str());
+
+    address.sun_family = AF_UNIX;
+#pragma warning(push)
+#pragma warning(disable:4996)
+    strncpy(address.sun_path, rootAddress.c_str(), sizeof(address.sun_path));
```

Comment:
Could you use strncpy_s so that way we don't have to suppress the warning. Then on Unix, we can add a macro that replaces strncpy_s with strncpy?